lib/rules/prevent-unused-connect-functions.js

```diff
@@ -0,0 +1,31 @@
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Components should only be wrapped in a connect function when state or dispatch is being used',
+    },
+    fixable: false,
+  },
+
+  create(context) {
+    function nullLiteral(node) {
+        return node.type === 'Literal' && node.value === null;
+    }
+
+    function arrowFunctionWithUnusedParams(node) {
+        return node.type === 'ArrowFunctionExpression' && node.params.length === 0;
+    }
+
+    return {
+      'ExportDefaultDeclaration CallExpression[callee.name=connect][arguments]'(node) {
+        if (node && node.arguments && node.arguments.length > 2) {
+            const stateArgument = node.arguments[0];
+            const dispatchArgument = node.arguments[1];
+            
+            if ((nullLiteral(stateArgument) || arrowFunctionWithUnusedParams(stateArgument)) && (nullLiteral(dispatchArgument) || arrowFunctionWithUnusedParams(dispatchArgument))) {
```

Comment:
I had a question typed up here that I think I forgot to finalize/post.

Does having an empty state or dispatch but not the other get caught here?  Is it even a valid edge case to consider?

```javascript
export default connect(
  (state) => ({}),
  () => ({}),
  null,
  { withRef: true }
)();
```


```javascript
export default connect(
  () => ({}),
  (dispatch) => ({}),
  null,
  { withRef: true }
)();
```

Reply:
It's acceptable to use just state or dispatch, we do this in many places.